api/redhatopenshift/HcpCluster/hcpCluster-models.tsp

```diff
@@ -474,6 +478,18 @@ model NodePoolSpec {
   tuningConfigs?: string[];
 }
 
+/** Taint is controlling the node taint and its effects */
+model Taint {
```

Comment:
The "key" attribute and the "effect" attribute should be required and not empty.

"value" is also required but it is not clear to me right now whether it can be empty or not at CS level. I will check this in CS doing some tests. For this specific attribute, if we need to change it we can do it in a follow-up MR

Reply:
I made only value optional, I will make the min length 3, as this is the smallest that makes sense for keys. This will force user to put value in the key.

Reply:
I am trying to understand this. Why is the minimum length of a label key 3? Reading the k8s doc it does not seem to mention it: https://kubernetes.io/docs/concepts/overview/working-with-objects/labels/#syntax-and-character-set